src/components/grid/grid-container/grid-container.component.js

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import GridContainerStyle from './grid-container.style';
+import GridItem from '../grid-item';
+
+const GridContainer = (props) => {
+  const {
+    children
+  } = props;
+  return (
+    <GridContainerStyle data-component='grid'>
```

Comment:
I think we should remove this component and use `GridContainerStyle` directly, renaming it to `GridContainer`.

Then we can change 
```diff
- GridContainerStyle = styled.div`
+ const Grid =({children, ...props}) => <div {...props} data-component='grid'>{children}</div>
+ Grid.propTypes = {
+  children: PropTypes.oneOfType([GridItem, PropTypes.arrayOf(GridItem)]).isRequired
+ }
+ GridContainer = styled(Grid)`
```
This would result in two components being rendered instead of three.
```diff
- GridContainer -> GridContainerStyle -> Styled(div)
+ GridContainer -> Styled(div)
```

Reply:
There is currently a bug in `addon-docs` which means that the documentation would not display any prop-types.

https://github.com/storybookjs/storybook/issues/8279

We have decided to keep the current approach, which is in line with existing components in Carbon.